src/waveform/waveform.h

```diff
@@ -31,6 +31,13 @@ union WaveformData {
 
 class Waveform {
   public:
+    enum class SaveState {
+        Virgin = 0,
+        Changing, 
```

Comment:
The state "Changing" is not directly used, but if we change the "Virgin" state to "Empty", then, what state would you set to a waveform that has m_complete > 0 and < m_datasize?  And don't tell me "SavePending", because that is precisely the one that we are trying to avoid. (except if then we need to do an atomic compare of both, savestate and complete to decide if it's time to save)